torch/utils/_triton.py

```diff
@@ -0,0 +1,17 @@
+import functools
+
+import torch
+
+from torch._inductor.cuda_properties import get_device_capability
+
+
+@functools.lru_cache(None)
+def has_triton() -> bool:
+    if not torch.cuda.is_available():
+        return False
+    try:
+        import triton
+
+        return triton is not None and get_device_capability() >= (7, 0)
+    except ImportError:
+        return False
```

Comment:
It would be nice to add `__all__` property to indicate that only `has_trition` should be used from this package.